src/algebra/big_operators/basic.lean

```diff
@@ -1718,6 +1718,48 @@ begin
   { rw [finset.sum_insert hat, finset.prod_insert hat, multiset.prod_add, ih] }
 end
 
+lemma disjoint_list_sum_right {a : multiset α} {l : list (multiset α)}
+  (h : ∀ b ∈ l, multiset.disjoint a b) :
+  multiset.disjoint a l.sum :=
+begin
+  classical,
+  induction l with b bs ih,
+  { exact (zero_disjoint _).symm },
+  { rw [list.sum_cons, disjoint_add_right],
+    exact ⟨h _ (list.mem_cons_self _ _), ih $ λ b hb, h _ $ list.mem_cons_of_mem _ hb⟩ }
+end
+
+lemma disjoint_sum_right {a : multiset α} {i : multiset (multiset α)} :
+  (∀ b ∈ i, multiset.disjoint a b) → multiset.disjoint a i.sum :=
+begin
+  classical,
+  induction i using quotient.induction_on,
+  intros h,
+  rw [quot_mk_to_coe, multiset.coe_sum],
+  exact disjoint_list_sum_right h,
+end
+
+lemma disjoint_finset_sum_right {β : Type*} {i : finset β} {f : β → multiset α} {a : multiset α}
+  (h : ∀ b ∈ i, multiset.disjoint a (f b)) :
+  multiset.disjoint a (∑ x in i, f x) :=
+by { classical, exact disjoint_sum_right (by simpa using h), }
+
+lemma finset_sum_eq_sup_of_disjoint {β : Type*} {i : finset β}
+  {f : β → multiset α} (h : ∀ x y ∈ i, x ≠ y → multiset.disjoint (f x) (f y)) :
+  ∑ x in i, f x = i.sup f :=
+begin
+  classical,
+  induction i using finset.induction_on with z i hz hr,
+  { simp only [finset.sum_empty, finset.sup_empty, bot_eq_zero], },
+    rw [finset.sum_insert hz, finset.sup_insert],
+    convert add_eq_union_iff_disjoint.mpr _,
+    exact (hr (λ x hx y hy hxy, h x (finset.mem_insert_of_mem hx)
+      y (finset.mem_insert_of_mem hy) hxy)).symm,
+    exact disjoint_finset_sum_right
+      (λ x hx, h z (finset.mem_insert_self z i) x (finset.mem_insert_of_mem hx)
+      (by {contrapose! hx, simp [←hx, hz], } : z ≠ x)),
+end
```

Comment:
Can you split these out into a new PR? It would be good to have the `left` versions of the disjoint lemmas too; they should follow trivially in term-mode from the `right` versions via `disjoint.symm`.

Reply:
Ok, this is now #15556